validators.js

```diff
@@ -46,3 +46,28 @@ export const isPercentage = value =>
 
 export const isCountZero = value =>
   value === 0 ? i18n.t('errors.moreThanZero') : undefined
+
+
+export const isFourDigitsNumber = val => {
+  if (!val || val?.length !== 4 || isNaN(val)) {
+    return i18n.t('errors.requireFourDigits')
+  }
+  else
+    return undefined
+}
+
+export const isEightDigitsNumber = val => {
+  if (!val || val?.length !== 8 || isNaN(val)) {
+    return i18n.t('errors.requireEightDigits')
+  }
+  else
+    return undefined
+}
```

Comment:
Can we do this in a more generic function like checkNumberLength(val, numLength)?

If so, your error localized error message can also be more generic and take in a parameter.